Inc/DirectXMathMatrix.inl

```diff
@@ -297,13 +297,20 @@ inline XMMATRIX XM_CALLCONV XMMatrixMultiply
     return mResult;
 #elif defined(_XM_SSE_INTRINSICS_)
     XMMATRIX mResult;
+    // Splat the component X,Y,Z then W
+#if defined(_XM_AVX_INTRINSICS_)
+    XMVECTOR vX = _mm_broadcast_ss(reinterpret_cast<const float*>(&M1.r[0]) + 0);
```

Comment:
I tried to keep a const float* to M1.r[0] and offset from 0 .. 15 with it but it actually crashed the VS2015 compiler/linker... I reported the issue with a test case and in the meantime, we can live with the ugly casting I suppose...